src/h5web/vis-packs/core/matrix/Cell.tsx

```diff
@@ -3,6 +3,21 @@ import type { GridChildComponentProps } from 'react-window';
 import { format } from 'd3-format';
 import styles from './MatrixVis.module.css';
 import { GridSettingsContext } from './GridSettingsContext';
+import type { Primitive } from '../../../providers/models';
+import type { PrintableType } from '../models';
+import { formatComplexValue } from '../../../utils';
+
+function formatPrintableValue(value: Primitive<PrintableType>) {
+  if (Array.isArray(value)) {
+    return formatComplexValue(value, '.1f');
+  }
+
+  if (typeof value === 'number') {
+    return format('.3e')(value);
+  }
+
+  return value.toString();
+}
```

Comment:
There is no `utils.ts` for `matrix`. So either this or nested ternaries...

Reply:
I reckon add the `utils.ts` 😉

Reply:
Just a thought for the future. Maybe the formatter should be passed by `MappedMatrixVis`. This way, we could pass the correct formatter for the dataset based on its type without having to check the value. `Array.isArray(value)` may be problematic in the future if other data types are stored as arrays, for instance.